tests/unit/test_application.py

```diff
@@ -143,3 +144,21 @@ def test_build_planner_success_architecture_all(base, build_base, expected_base)
         assert build_info.platform == f"{expected_base.name}@{expected_base.version}"
 
     assert "all" not in [a.build_on for a in architectures]
+
+
+@pytest.mark.parametrize("new_envvar", {"CRAFT_BUILD_FOR", "CRAFT_BUILD_ENVIRONMENT"})
```

Comment:
maybe we should put these variables in a top-level list in `snapcraft.application` and then use it both there and here

Reply:
@tigarmo How about something like this in the top level of `snapcraft/application.py`?

```
MAPPED_ENV_VARS = {
    ev: "SNAP" + ev for ev in ("CRAFT_BUILD_FOR", "CRAFT_BUILD_ENVIRONMENT")
}   
```